letsencrypt/client/revoker.py

```diff
@@ -6,146 +14,513 @@
+            return [str(self.idx), self.orig.path, self.orig_key.path]
+        return None
+
+    def add_meta(self, idx, orig, orig_key, backup, backup_key):
+        """Add meta data to cert
+
+        :param int idx: convenience index for revoker
+        :param tuple orig: (`str` original certificate filepath, `str` status)
+        :param tuple orig_key: (`str` original auth key path, `str` status)
+        :param str backup: backup certificate filepath
+        :param str backup_key: backup key filepath
+
+        """
+        status = ""
+        key_status = ""
+
+        # Verify original cert path
+        if not os.path.isfile(orig):
+            status = Cert.DELETED_MSG
+        else:
+            o_cert = M2Crypto.X509.load_cert(orig)
+            if self.get_fingerprint() != o_cert.get_fingerprint(md="sha1"):
+                status = Cert.CHANGED_MSG
+
+        # Verify original key path
+        if not os.path.isfile(orig_key):
+            key_status = Cert.DELETED_MSG
+        else:
+            with open(orig_key, "r") as fd:
+                key_pem = fd.read()
+            with open(backup_key, "r") as fd:
+                backup_key_pem = fd.read()
+            if key_pem != backup_key_pem:
+                key_status = Cert.CHANGED_MSG
+
+        self.idx = idx
+        self.orig = Cert.PathStatus(orig, status)
+        self.orig_key = Cert.PathStatus(orig_key, key_status)
+        self.backup_path = backup
+        self.backup_key_path = backup_key
+
+    def get_installed_msg(self):
+        """Access installed message."""
+        return ", ".join(self.installed)
+
+    def get_subject(self):
+        """Get subject."""
+        return self.cert.get_subject().as_text()
+
+    def get_cn(self):
+        """Get common name."""
+        return self.cert.get_subject().CN
+
+    def get_issuer(self):
+        """Get issuer."""
+        return self.cert.get_issuer().as_text()
+
+    def get_fingerprint(self):
+        """Get sha1 fingerprint."""
+        return self.cert.get_fingerprint(md="sha1")
+
+    def get_not_before(self):
+        """Get not_valid_before field."""
+        return self.cert.get_not_before().get_datetime()
+
+    def get_not_after(self):
+        """Get not_valid_after field."""
+        return self.cert.get_not_after().get_datetime()
+
+    def get_serial(self):
+        """Get serial number."""
+        self.cert.get_serial_number()
+
+    def get_pub_key(self):
+        """Get public key size."""
+        # .. todo:: M2Crypto doesn't support ECC, this will have to be updated
+        return "RSA " + str(self.cert.get_pubkey().size() * 8)
+
+    def get_san(self):
+        """Get subject alternative name if available."""
+        try:
+            return self.cert.get_ext("subjectAltName").get_value()
+        except LookupError:
+            return ""
+
+    def __str__(self):
+        text = []
+        text.append("Subject: %s" % self.get_subject())
```

Comment:
instead of `text.append` just define it in the datatype constructor:
```python
text = [
    "Subject: %s" % self.cert.get_subject().as_text(),
    ...
]

readability++, performance++